src/coord/src/catalog/builtin.rs

```diff
@@ -1189,6 +1202,28 @@ lazy_static! {
             Builtin::View(&PG_RANGE),
             Builtin::View(&PG_ENUM),
         ];
+
+        // TODO(sploiselle): assign static global IDs to functions
+        let mut func_global_id_counter = 2000;
+
+        for (schema, funcs) in &[
+            (PG_CATALOG_SCHEMA, &*sql::func::PG_CATALOG_BUILTINS),
+            (MZ_CATALOG_SCHEMA, &*sql::func::MZ_CATALOG_BUILTINS),
+            (MZ_INTERNAL_SCHEMA, &*sql::func::MZ_INTERNAL_BUILTINS),
+        ] {
+            for (name, func) in funcs.iter() {
+                builtins.push(Builtin::Func(BuiltinFunc {
+                    name,
+                    schema,
+                    id: GlobalId::System(func_global_id_counter),
+                    inner: func,
+                }));
+                func_global_id_counter += 1;
+            }
+        }
+
+        assert!(func_global_id_counter < 3000, "exhausted func global IDs");
```

Comment:
Oh interesting, do you think it'd be worth adding something similar for other `Builtin` variants?

Reply:
This is the only builtin variant that dynamically assigns `GlobalId`s on start; everything else has them statically assigned. This one needs to do it, as well––I just have been lazy in terms of generating/assigning them.